packages/legend-application-studio/src/components/editor/edit-panel/connection-editor/RelationalDatabaseConnectionEditor.tsx

```diff
@@ -1151,6 +1135,218 @@ const RelationalConnectionStoreEditor = observer(
+              observerContext,
+              new MapperPostProcessor(),
+            );
+            break;
+          }
+          default: {
+            const extraPostProcessorCreators = plugins.flatMap(
+              (plugin) =>
+                (
+                  plugin as StoreRelational_LegendStudioApplicationPlugin_Extension
+                ).getExtraPostProcessorCreators?.(
+                  connectionValueState,
+                  observerContext,
+                ) ?? [],
+            );
+            for (const creator of extraPostProcessorCreators) {
+              creator(postProcessorType);
+            }
+          }
+        }
+
+        connectionValueState.postProcessorState?.setPostProcessorState(
+          connectionValueState.connection.postProcessors[
+            connectionValueState.connection.postProcessors.length - 1
+          ],
+        );
+      };
+
+    const selectPostProcessor = (postProcessor: PostProcessor): void => {
+      connectionValueState.selectPostProcessor(postProcessor);
+    };
+
+    return (
+      <div className="panel__container ">
+        <ResizablePanelGroup orientation="horizontal">
+          <ResizablePanelSplitter />
+          <ResizablePanel>
+            <div className="panel__container panel__container--border-top">
+              <ResizablePanelGroup orientation="vertical">
+                <ResizablePanel size={150} minSize={70}>
+                  <div className="panel__container panel__container--border-right">
+                    <DropdownMenu
+                      content={postProcessorOptions.map((postProcessorType) => (
+                        <MenuContentItem
+                          key={postProcessorType.value}
+                          onClick={addPostProcessor(postProcessorType.value)}
+                        >
+                          New {postProcessorType.label} Post Processor
+                        </MenuContentItem>
+                      ))}
+                      menuProps={{
+                        anchorOrigin: {
+                          vertical: 'bottom',
+                          horizontal: 'right',
+                        },
+                        transformOrigin: {
+                          vertical: 'top',
+                          horizontal: 'right',
+                        },
+                        elevation: 7,
+                      }}
+                    >
+                      <PanelHeader title="post-processor">
+                        <PanelHeaderActionItem tip="Create Post-Processor">
+                          <PlusIcon />
+                        </PanelHeaderActionItem>
+                      </PanelHeader>
+                    </DropdownMenu>
+                    <div className="panel__content">
+                      {postProcessors.map((postProcessor, idx) => (
+                        <ContextMenu
+                          key={postProcessor._UUID}
+                          disabled={false}
```

Comment:
Looks to me like we haven't account for `read-only` case when people can't really edit this connection. Look into how we use `isReadOnly` flag

Reply:
hmm yes i showed blatant disregard for the read-only...adding 👍